config.yaml

```diff
@@ -1,14 +1,42 @@
-# This file defines charm config options, and populates the Configure tab on Charmhub.
-# If your charm does not require configuration options, delete this file entirely.
-#
-# See https://juju.is/docs/config for guidance.
-
 options:
-  role:
-    description: Role for this microk8s deployment
-    default: ""
+  addons:
+    description: |
+      Space-separated list of addons that should be enabled and managed by the charm. Removing
+      an addon from this list will disable it in the cluster.
+
+      See https://microk8s.io/docs/addons for a comprehensive list of addons available in MicroK8s.
+
+      Use "addon:arg1:arg2" to specify arguments when enabling a specific addon.
+
+      Examples:
+      - "dns rbac hostpath-storage ingress"
+      - "dns rbac hostpath-storage ingress gpu:--version=v1.10.1"
+
+    default: "dns rbac hostpath-storage ingress"
     type: string
   channel:
-    description: MicroK8s channel to install
+    description: |
```

Comment:
This config option should be removed -- this should be controlled by the channel of the charm, not through config

Reply:
This is to facilitate two requirements:

- picking a flavored branch (e.g. 1.27 vs 1.27-strict vs 1.27-eksd)
- Controlled upgrades to new versions (upgrade charm track -> update channel config -> run upgrade action).

This is similar to what Charmed Kubernetes is doing as well.

Reply:
The correct way to handle this is using tracks/channels in the store. The pattern that Charmed Kubernetes uses (that was also popular in Openstack previously) came from a time where we didn't have the correct store semantics for versioning.

Any charm upgrades should be handled using the `juju refresh` mechanics, not through actions. Any logic that would be required in that action should be moved into the `upgrade` hook handler.

If you need to create tracks/channels in the charm store that mirror those of the snap, then that's fine :)

Reply:
Dealing with this in #81